### PR TITLE
FEATURE: upgrade k8s to 1.27.4 and cilium to 1.14.2

### PR DESCRIPTION
**Pull request by imo-ininder**

### CODE REVIEW

File: templates/network-plugins/cilium/cilium-vxlan.yaml.tpl [hunk 1]
```diff
@@ -746,16 +664,21 @@ spec:
...
+        container.apparmor.security.beta.kubernetes.io/apply-sysctl-overwrites: "unconfined"
       labels:
         k8s-app: cilium
         app.kubernetes.io/name: cilium-agent
         app.kubernetes.io/part-of: cilium
     spec:
       containers:
       - name: cilium-agent
-        image: "${agent_repo}:${agent_tag}"
+        image: "quay.io/cilium/cilium:v1.14.2"
```
Comment (soem) [inline templates/network-plugins/cilium/cilium-vxlan.yaml.tpl:681]: Hi @imo-ininder 
I found that image version changes have been applied on `variables_defaults.tf`.
I'd think we should keep `"${agent_repo}:${agent_tag}"` if it works.

**Review by soem — CHANGES_REQUESTED** [review]

**Review by Downager — CHANGES_REQUESTED** [review]
vxlan-values.yaml 有很多值是有根據需求調整跟換成 template var，升級時不能直接用 default values.yaml 覆蓋掉

**Comment by imo-ininder** [comment]
已經根據原本的需求重新調整成 合適的 vxlan-values.yaml 並重新產生 cilium-vxlan.yaml.tpl

**Review by soem — APPROVED** [review]

File: templates/network-plugins/cilium/vxlan-values.yaml [hunk 2]
```diff
@@ -1445,7 +1578,7 @@ readinessProbe:
   periodSeconds: 30
 
 # -- Configure the kube-proxy replacement in Cilium BPF datapath
-# Valid options are "disabled", "partial", "strict".
+# Valid options are "true", "false", "disabled" (deprecated), "partial" (deprecated), "strict" (deprecated).
 # ref: https://docs.cilium.io/en/stable/network/kubernetes/kubeproxy-free/
 kubeProxyReplacement: "strict"
```
Comment (Downager) [inline templates/network-plugins/cilium/vxlan-values.yaml:1583]: Valid options are "true", "false", "disabled" (deprecated), "partial" (deprecated), "strict" (deprecated).

要改成 true

**Review by Downager — CHANGES_REQUESTED** [review]

**Review by Downager — APPROVED** [review]

**Pull request merged**